eng/pipelines/cpp-nightly.yml

```diff
@@ -0,0 +1,64 @@
+
+jobs:
+  - job:
+    displayName: Nightly vcpkg PR
+    pool:
+      name: azsdk-pool-mms-win-2019-general
+      vmImage: MMS2019
+
+    variables:
+      # The branch in the azure-sdk/vcpkg repo that is updated nigthly with the
+      # latest set of ports
+      TargetPrBranch: azure-sdk-for-cpp-nightly-verification
+
+    steps:
+      - checkout: self
+
+      - template: /eng/pipelines/templates/steps/generate-nightly-branch-name.yml
+
+      - template: /eng/pipelines/templates/steps/vcpkg-clone.yml
+        parameters:
+          Workspace: $(Pipeline.Workspace)
+          RepoOwner: azure-sdk
+          PRBranchName: $(TargetPrBranch)
+
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        parameters:
+          WorkingDirectory: $(Pipeline.Workspace)/vcpkg
+
+      # Git merge does not support the -c parameter. For the scope of this job
+      # it's ok to configure the user.name and user.email at a global level
+      - pwsh: |
```

Comment:
We have generally been avoiding setting these globally as it might indirectly impact any other commits that happen in this job. That is why we try to pass them as a config option to the commit itself.

Reply:
There are some cases where `git merge` will create a merge commit (even when `--no-commit` is specified). However I can use `git config --unset` to remove the config after the commit steps have completed and keep the environment clean.

Reply:
I'd still like to better understand why we are doing a merge over a rebase for this nightly branch, however even if we keep doing the merge you can still pass the config info to the merge command similar to what we do for the commit to prevent the need for setting and unsetting the values.

Reply:
Actual process: 
1. `git checkout $(NightlyBranchName)`
2. Set `HasChanges` to `true` 
3. All done